pyscriptjs/src/exceptions.ts

```diff
@@ -29,34 +24,28 @@ export enum ErrorCode {
 }
 
 export class UserError extends Error {
-    messageType: MessageType;
-    errorCode: ErrorCode;
     /**
      * `isinstance` doesn't work correctly across multiple realms.
      * Hence, `$$isUserError` flag / marker is used to identify a `UserError`.
      */
     $$isUserError: boolean;
 
-    constructor(errorCode: ErrorCode, message: string, t: MessageType = 'text') {
+    constructor(public errorCode: ErrorCode, message: string, public messageType: MessageType = 'text') {
         super(message);
-        this.errorCode = errorCode;
```

Comment:
Will this still make errorCode be required?

Reply:
it's required by the TS ignature and assigned automatically as own public property

Reply:
uh wait ... it should indeed probably be passed to the super, I'll have a look

Reply:
Okay for context when we implemented the error code the idea was that we should always require a code when raising an exception.

One thing we haven't been doing much is linking to the right docs haha

Reply:
P.S. at the same time we can move `this.message = ...` to the super call because we inherit the `message` property so it was me overlooking at that override that is not necessary ... will fix the code, **and** the code is indeed part of the message: my bad!

Reply:
But we use the `errorCode` in the message shown to the user and logs right?

Reply:
yes, that's correct ... I was confused by `super(message)` that we were replacing right after with `this.messaage = ...` so **now** we just use `super(`(${errorCode}) ${message}`)` which attaches automatically the `message` property because that's what `Event` does and TS attaches the `this.errorCode` automatically because that's how constructors with `public` works in TS ... we should be good now, it's exactly same behavior as before but there's less code to maintain.

Reply:
Ah gotcha that sounds good I was mostly asking just to know if the behaviour changed haha 

Less good seems like a good idea 😄